src/mono/mono/mini/debugger-engine.c

```diff
@@ -1323,8 +1323,11 @@ mono_de_ss_start (SingleStepReq *ss_req, SingleStepArgs *ss_args)
 	} else {
 		frame_index = 1;
 
+#ifndef TARGET_WASM
```

Comment:
Why doesn't wasm have ctx? The type seems to be defined as:
```c
    #if defined (TARGET_WASM)
   
    typedef struct {
            host_mgreg_t wasm_sp;
            host_mgreg_t wasm_bp;
            host_mgreg_t llvm_exc_reg;
            host_mgreg_t wasm_ip;
            host_mgreg_t wasm_pc;
    } MonoContext;
```
.. with `wasm_*` fields. Are they not relevant here?

If it is expected to be NULL always, then should we add an assert in `mini_wasm_debugger.c:ss_calculate_framecount` for `ctx`, so we know if that changes in the future?

Reply:
We get the CTX calling the `rt_callbacks.tls_get_restore_state (tls)` and on wasm this returns NULL. And for `ss_calculate_framecount` we don't use `ctx` information on wasm, that is why I just removed it from the if.
I don't think we should assert, because we will not use it for nothing now. Even if it has value, it will not break anything.